tx.go

```diff
@@ -317,7 +317,11 @@ func (tx *Tx) WriteTo(w io.Writer) (n int64, err error) {
 	if err != nil {
 		return 0, err
 	}
-	defer func() { _ = f.Close() }()
+	defer func() {
+		if err != nil {
```

Comment:
This will still double close when `return n, f.Close()` has an error. Try `return n, nil` and

```go
defer func() {
    if cerr := f.Close(); err == nil {
        err = cerr
    }
}
```